src/main/java/com/cryptomorin/xseries/SkullUtils.java

```diff
@@ -312,7 +312,10 @@ public static StringSkullCache detectSkullValueType(@Nonnull String identifier)
         }
 
         if (isUsername(identifier)) return new StringSkullCache(ValueType.NAME);
-        if (identifier.contains("textures.minecraft.net")) return new StringSkullCache(ValueType.TEXTURE_URL);
+        if (identifier.contains("textures.minecraft.net")) {
+            String mojangSHA = identifier.substring(identifier.lastIndexOf('/'));
```

Comment:
You should fix `extractMojangSHAFromBase64()` instead of just taking a substring.